eclipse-theia/theia-blueprint
# Theia 1.53.0

#### What it does

* update to latest Theia (#378 on windows)
* update electron
* add @theia/collaboration #383
* update Jenkinsfile global environment
* create (optional) desktop file for linux desktop integration (#378 on linux)
  * I think for testing you might have to refresh with `update-desktop-database ~/.local/share/applications/` after the desktop file was created
* add protocol for mac (#378 on mac)

closes #378 
closes #383

Contributed on behalf of STMicroelectronics

#### How to test
* test that IDE is still working as expected
* test collaboration feature and that info text is available on welcome page
* test `theia://` protocol handling (see https://github.com/eclipse-theia/theia/pull/13306 for instructions)
  * on Linux test that opting out of desktop file creation works as well
    * the file that stores user decisions is located at `~/.theia-ide/globalStorage/theia-ide-launcher/desktopfile.json`
    * current behavior:
      * we only maintain one set of Theia IDE desktop files. If there are two TheiaIDE AppImages in different locations only one can be set. (Users may create them themselves if required)
      * we store if an AppImage path should not have a desktop file -> the user will not be asked again
        * this is stored in the declined array in above file
      * if an AppImage path is not declined we check whether it is the current default
        * if yes -> no op since already set
        * if no -> ask user if desktop files should be created/updated

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
